subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -856,9 +860,24 @@ static int start_job(struct nrf_cloud_fota_job *const job)
 	__ASSERT_NO_MSG(job != NULL);
 	int ret;
 
-	ret = fota_download_start(job->info.host, job->info.path,
-		CONFIG_NRF_CLOUD_SEC_TAG, NULL,
-		CONFIG_NRF_CLOUD_FOTA_DOWNLOAD_FRAGMENT_SIZE);
+	enum dfu_target_image_type img_type;
+
+	switch (job->info.type) {
+	case NRF_CLOUD_FOTA_BOOTLOADER:
+	case NRF_CLOUD_FOTA_APPLICATION:
+		img_type = DFU_TARGET_IMAGE_TYPE_MCUBOOT;
+		break;
+	case NRF_CLOUD_FOTA_MODEM:
+		img_type = DFU_TARGET_IMAGE_TYPE_MODEM_DELTA;
+	default:
+		LOG_ERR("Unhandled FOTA type: %d", job->info.type);
```

Comment:
Would it make sense to support non-standard types by passing unrecognized ones to the original `fota_download_start`?